homeassistant/components/emulated_hue/hue_api.py

```diff
@@ -251,106 +278,158 @@ def put(self, request, username, entity_number):
             json_response.append(create_hue_success_response(
                 entity_id, HUE_API_STATE_BRI, brightness))
 
+        if color is not None:
+            json_response.append(create_hue_success_response(
+                entity_id, HUE_API_STATE_XY, color))
+
+        if color_temp is not None:
+            json_response.append(create_hue_success_response(
+                entity_id, HUE_API_STATE_CT, color_temp))
+
         return self.json(json_response)
 
 
 def parse_hue_api_put_light_body(request_json, entity):
     """Parse the body of a request to change the state of a light."""
+    result = None
+    brightness = None
+    color = None
+    color_temp = None
+
     if HUE_API_STATE_ON in request_json:
         if not isinstance(request_json[HUE_API_STATE_ON], bool):
             return None
 
-        if request_json['on']:
-            # Echo requested device be turned on
-            brightness = None
-            report_brightness = False
-            result = True
-        else:
-            # Echo requested device be turned off
-            brightness = None
-            report_brightness = False
-            result = False
+        result = request_json['on']
+
+    # Make sure the entity actually supports color temperature
+    entity_features = entity.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
 
     if HUE_API_STATE_BRI in request_json:
         try:
             # Clamp brightness from 0 to 255
-            brightness = \
+            hue_brightness = \
                 max(0, min(int(request_json[HUE_API_STATE_BRI]), 255))
         except ValueError:
             return None
 
-        # Make sure the entity actually supports brightness
-        entity_features = entity.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
-
         if entity.domain == "light":
             if entity_features & SUPPORT_BRIGHTNESS:
-                report_brightness = True
+                brightness = hue_brightness
                 result = (brightness > 0)
 
         elif (entity.domain == "script" or
               entity.domain == "media_player" or
               entity.domain == "fan"):
             # Convert 0-255 to 0-100
-            level = brightness / 255 * 100
+            level = hue_brightness / 255 * 100
             brightness = round(level)
-            report_brightness = True
             result = True
 
-    return (result, brightness) if report_brightness else (result, None)
+    if HUE_API_STATE_XY in request_json:
+        if not isinstance(request_json[HUE_API_STATE_XY], list) or \
+                len(request_json[HUE_API_STATE_XY]) != 2:
+            return None
+
+        if entity.domain == "light":
```

Comment:
This requires testing. https://coveralls.io/builds/10389271/source?filename=homeassistant%2Fcomponents%2Femulated_hue%2Fhue_api.py#L334